### PR TITLE
Add cost estimation

### PR DESCRIPTION
**Pull request by alexander-veit**
This PR adds cost estimation to Tibanna workflow runs. Estimated costs can be obtained as soon as the job finished. It is available from the command line by invoking `cost_estimate`. When the true costs are available, `cost_estimate` will return the true costs instead of the estimate.
The estimated costs will also be automatically added to the metrics report.

The estimated costs seem accurate, when the true price is more than a few cent. In other words, the relative error is large, when the true cost is below 1 or 2 cent, otherwise the relative error is small.

### CODE REVIEW

File: tibanna/cw_utils.py [hunk 1]
```diff
@@ -615,6 +623,10 @@ def create_html(cls):
                         <td class="left">Cost</td>
                         <td class="center">%s</td>
                       </tr>
+                      <tr>
+                        <td class="left">Cost (estimated)</td>
```
[the remark this answers is not in the document]
Comment (alexander-veit) [reply tibanna/cw_utils.py]: Done

File: tibanna/core.py [hunk 2]
```diff
@@ -1072,16 +1113,31 @@ def reformat_time(t, delta):
...
         cost = sum([float(_['Total']['BlendedCost']['Amount']) for _ in billingres['ResultsByTime']])
         if update_tsv:
             log_bucket = postrunjson.config.log_bucket
             # reading from metrics_report.tsv
             does_key_exist(log_bucket, job_id + '.metrics/metrics_report.tsv')
             read_file = read_s3(log_bucket, os.path.join(job_id + '.metrics/', 'metrics_report.tsv'))
-            if 'Cost' not in read_file:
-                write_file = read_file + 'Cost\t' + str(cost) + '\n'
+
+            # Remove Estimated_cost from file, because we want tp update it
```
[the remark this answers is not in the document]
Comment (alexander-veit) [reply tibanna/core.py]: I added the option `overwrite_cost_estimate_in_tsv`.

**Comment by alexander-veit** [comment]
Ok, I think this is better now.
- `cost --update-tsv` will update only the "Cost" row in the tsv. It will always be updated (overwritten), even if it was there before.
- `cost_estimate --update-tsv` will update only the "Cost_Estimate" row in the tsv. It will always be updated (overwritten), even if it was there before. In case the exact cost is available, `cost_estimate` will return the exact cost and update the "Cost_Estimate" row in the tsv with that value. The "Cost" row is not updated automatically in that case. `cost --update-tsv` needs to be called for that.

Does that make sense?

**Comment by SooLee** [comment]
Yup that sounds good.

**Pull request merged**